src/SonarScanner.MSBuild.PreProcessor/WebService/SonarWebService.cs

```diff
@@ -240,45 +161,45 @@ public Task<IList<SensorCacheEntry>> DownloadCache(string projectKey, string bra
             return downloader.DownloadStream(uri).ContinueWith(ParseCacheEntries);
         }
 
-        public async Task<bool> IsSonarCloud() =>
-            SonarProduct.IsSonarCloud(serverUri.Host, await GetServerVersion());
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
 
-        private IList<SensorCacheEntry> ParseCacheEntries(Task<Stream> task)
+        protected virtual void Dispose(bool disposing)
         {
-            if (task.IsFaulted || task.Result is not { } dataStream)
+            if (disposed)
             {
-                return new List<SensorCacheEntry>();
+                return;
             }
 
-            var cacheEntries = new List<SensorCacheEntry>();
-            try
+            if (disposing)
             {
-                while (dataStream.Position < dataStream.Length)
-                {
-                    cacheEntries.Add(SensorCacheEntry.Parser.ParseDelimitedFrom(dataStream));
-                }
-            }
-            catch (Exception e)
-            {
-                logger.LogDebug(Resources.MSG_IncrementalPRCacheEntryDeserialization, e.Message);
+                downloader.Dispose();
             }
 
-            return cacheEntries;
+            disposed = true;
         }
 
-        private async Task<Uri> AddOrganization(Uri uri, string organization)
+        /// <summary>
+        /// Retrieves project properties from the server.
+        ///
+        /// Will fail with an exception if the downloaded return from the server is not a JSON array.
+        /// </summary>
+        /// <param name="projectKey">The SonarQube project key to retrieve properties for.</param>
+        /// <param name="projectBranch">The SonarQube project branch to retrieve properties for (optional).</param>
+        /// <returns>A dictionary of key-value property pairs.</returns>
+        ///
+        public async Task<IDictionary<string, string>> GetProperties(string projectKey, string projectBranch)
```

Comment:
I'd like to rename all `Get` that makes HTTP calls to `Download` in this file. To unhide their real functionality. Probably in another PR:
* GetRules
* GetAllLanguages
* GetProperties

Reply:
We can do this in another PR otherwise rebase will be unnecessary hard :)